### PR TITLE
Claudia Airport Challenge

### PR DESCRIPTION
**Pull request by claudiagreen**

### CODE REVIEW

**Comment by mechacoach** [comment]
You had your airport_challenge reviewed by **EIrik**

### The good points are:

* `context` or `describe` blocks are used (where appropriate) to group tests
* The solution avoids vacuous tests
* There are a minimal number of untested paths in the code
* Unit tests all have single expectations
* Random behaviour is stubbed where appropriate
* The code follows Ruby naming conventions, Readability is supported by consistent naming
* Guard clauses are used where appropriate
* Each methods has a single responsibility
* Code avoids strings where it should use symbols
* Command/Query naming is not an issue in the code

### You should consider the following improvements:

* README should be updated
* Installation Instructions should be in the README
* Use named subject with `described_class`
* Use `before` blocks to set up objects rather than repeat code
* Get rid of redundant `respond_to` expectations
* Remove all commented-out code
* Do not expose internal implementation
* Avoid magic numbers (e.g. on capacity)
* Improper use of attr_*
* Avoid redundant lines of code


see https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md for more details

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  def instruct_takeoff(plane)
+    fail "Airport empty." if empty?
+    fail "Storm." if storm?
+    fail "That plane isn't in that airport." if plane_in_airport(plane) == false
+    if plane.landed?
+      plane.takeoff
+      plane = @planes.pop
+      @planes.delete(plane)
+    else
+      "Plane has taken off."
```
Comment (silvabox) [inline lib/airport.rb]: What does it mean to return a string from this method?

File: lib/airport.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  def instruct_land(plane)
+    fail "Airport full." if full?
+    fail "Storm." if storm?
+    unless plane.landed?
+      plane.land
+      @planes << plane
+      plane
+    else
+      "Plane has landed."
```
Comment (silvabox) [inline lib/airport.rb]: What does it mean to return a string from this method?

**Comment by silvabox** [comment]
Well done Claudia - that's a pretty good effort.  You should avoid exposing your planes array with an `attr_reader`

**Pull request closed without merging**